### PR TITLE
Set HeadCommit when creating tags

### LINKED ISSUE #18112: Tag push creates webhook with empty `commits` and `head_commit`
### Gitea Version

1.15.8

### Git Version

2.33.1

### Operating System

Windows Server 2016

### How are you running Gitea?

based on gitea-1.15.8-windows-4.0-amd64.exe as Windows Service

### Database

SQLite

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Description

Creating a new tag for a gitea repo creates a webhook with empty `commits` and `head_commit`. See example from gitea demo site:
```
{
  "ref": "refs/tags/v1.1.1",
  "before": "0000000000000000000000000000000000000000",
  "after": "3f150d39b92dd50cdd0248f5edea5767c2172584",
  "compare_url": "https://try.gitea.io/",
  "commits": [],
  "head_commit": null,
  "repository": {
  ...
```

I have an on-prem gitea instance connected with on-prem AppVeyor server. AppVeyor can't create new builds if  `commits` and `head_commit` are empty, see also open issue on AppVeyor: appveyor/ci#3765

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by KN4CK3R**
fixes #18112

It's the same behaviour as if you create a branch: empty commit list but the head commit is the last commit.

I'm not sure about the compare url. With `git.EmptySHA` you will see a 404 page. GitHub displays a "There isn’t anything to compare" page in this case.

### CODE REVIEW

**Comment by Gusted** [comment]
> I'm not sure about the compare url. With `git.EmptySHA` you will see a 404 page. GitHub displays a "There isn’t anything to compare" page in this case.

Well, that should be a enhancement for another PR as it's in this case a valid URL in that sense.

**Review by Gusted — APPROVED** [review]

File: services/release/release.go [hunk 1]
```diff
@@ -72,13 +72,17 @@ func createTag(gitRepo *git.Repository, rel *models.Release, msg string) (bool,
 			created = true
 			rel.LowerTagName = strings.ToLower(rel.TagName)
 
+			commits := repository.NewPushCommits()
```
Comment (wxiaoguang) [inline services/release/release.go:75]: It's surprising that `NewPushCommits` only creates an empty struct ....

And the code below duplicates. Could we introduce some more meaningful functions?

ps: I can accept that what the code does, if there is a chance, maybe we can clarify the code.

Comment (KN4CK3R) [reply services/release/release.go:75]: How would you deduplicate it? There may be a method which sets the head commit in the "constructor" but the compare url needs too many parameters. Feel free to change it.

Comment (wxiaoguang) [reply services/release/release.go:75]: How about:

```
func NewPushCommitsForNotification(commit) {
	commits := repository.NewPushCommits()
	commits.HeadCommit = repository.CommitToPushCommit(commit)
	commits.CompareURL = rel.Repo.ComposeCompareURL(git.EmptySHA, commit.ID.String())
	return commits
}
```

**Review by wxiaoguang — APPROVED** [review]

File: services/release/release.go (hunk 1 above)
Comment (KN4CK3R) [reply services/release/release.go:75]: But that name would be wrong for the "deleted ref" notification or `push.go@200`.

**Comment by lunny** [comment]
I think it's better to backport this to v1.15

**Comment by lunny** [comment]
Please send backport.

**Pull request merged**